src/soc/sifive/fu540/src/bootblock.S

```diff
@@ -26,6 +26,21 @@ _boot:
 	# some hardware configurations, but is always set in QEMU.
 	csrr a0, mhartid
 
+	# Reset the mtimecmp to some absurdly large number.
+	# TODO: Move to Rust
+	li t0, 0x100000000
+	li t1, 0x2004000
+	sd t0, (t1)
+	sd t0, 0x8(t1)
```

Comment:
can you put a comment here that you are setting it up for all the harts.

Reply:
I changed it so each hart sets its own mtimecmp value.